sysml.library/Kernel Library/Occurrences.kerml

```diff
@@ -67,38 +68,41 @@ package Occurrences {
 		 * Time slices of an occurrence that happen at a single instant of time
 		 * (i.e., have no duration).
 		 */
-		portion feature snapshots: Occurrence[0..*] subsets timeSlices;
+		portion feature all snapshots: Occurrence[1..*] subsets timeSlices;
+		inv { snapshots->forAll s:Occurrence (s::startShot == s::endShot) }
+		inv { snapshots == union(startShot, union(middleShots, endShot)) }
 		
 		/**
 		 * Occurrences of which this occurrence is a snapshot.
 		 */
-		feature snapshotOf : Occurrence[1..*] subsets timeSliceOf;
+		feature snapshotOf : Occurrence[0..*] subsets timeSliceOf;
 		
 		/**
 		 * The snapshot representing the start of the occurrence in time.
 		 */
 		portion feature startShot: Occurrence[1] subsets snapshots;
 		
-		/** 
-		 * The snapshot representing the end of the occurrence in time.
+		/**
+		 * The snapshots in between the startShot and endShot. There are none
+		 * when the startShot and endShot are the same.
 		 */
-		portion feature endShot: Occurrence[1] subsets snapshots;
+		portion feature all middleShots: Occurrence[0..*] subsets snapshots;
+		inv { isEmpty(middleShots) == (startShot == endShot) }
 		
-		/**
-		 * Snapshots have no duration.
+		/** 
+		 * The snapshot at the end of the occurrence in time.
 		 */
-		succession :Occurrences::HappensBefore, SelfLink 
-			first snapshots[1] then snapshots[1];
+		portion feature endShot: Occurrence[1] subsets snapshots;
 		
 		/**
-		 * The startShot happens before all snapshots.
+		 * The startShot happens before all middleShots.
 		 */
-		succession startShot[1] then snapshots[0..*];
```

Comment:
These are connector end multiplicities, right?   Is so, we're good, don't need isSufficient (all snapshot on the [0..*] will need to happensBefore/After the one one the [1] end).

Reply:
Yes these are connector end multiplicities.